File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/wrapping_and_braces/JavaKeepWhenReformattingSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavaKeepWhenReformattingSettings.class)
public interface JavaKeepWhenReformattingSettings {

  Optional<Boolean> lineBreaks();

  Optional<Boolean> commentAtFirstColumn();

  Optional<Boolean> controlStatementInOneLine();

  Optional<Boolean> multipleExpressionsInOneLine();

  Optional<Boolean> simpleBlocksInOneLine();

  Optional<Boolean> simpleMethodsInOneLine();

  Optional<Boolean> simpleLambdasInOneLine();

  Optional<Boolean> simpleClassesInOneLine();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/wrapping_and_braces/JavaKeepWhenReformattingSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.wrapping_and_braces

import com.fasterxml.jackson.annotation.JsonProperty

data class JavaKeepWhenReformattingSettings(

    @JsonProperty("Force braces")
    val lineBreaks: Boolean? = null,

    @JsonProperty("Comment at first column")
    val commentAtFirstColumn: Boolean? = null,

    @JsonProperty("Control statement in one line")
    val controlStatementInOneLine: Boolean? = null,

    @JsonProperty("Multiple expressions in one line")
    val multipleExpressionsInOneLine: Boolean? = null,

    @JsonProperty("Simple blocks in one line")
    val simpleBlocksInOneLine: Boolean? = null,

    @JsonProperty("Simple methods in one line")
    val simpleMethodsInOneLine: Boolean? = null,

    @JsonProperty("Simple lambdas in one line")
    val simpleLambdasInOneLine: Boolean? = null,

    @JsonProperty("Simple classes in one line")
    val simpleClassesInOneLine: Boolean? = null
)
